include/boost/math/special_functions/gamma.hpp

```diff
@@ -58,15 +58,15 @@ template <class T>
 inline bool is_odd(T v, const std::true_type&)
 {
    int i = static_cast<int>(v);
-   return i&1;
+   return i & 1 == 1;
```

Comment:
`==` has higher precedence than `&` so this is parsed as
```
return i & (1 == 1);
```
Personally I prefer the current version: `return i&1;`.